app/Libraries/OsuAuthorize.php

```diff
@@ -1135,6 +1135,21 @@ public function checkUserShow($user, $owner)
         }
     }
 
+    public function checkMatchView($user, $match)
+    {
+        if (!$match->private) {
+            return 'ok';
+        }
+
+        $this->ensureLoggedIn($user);
+
+        if (!$match->hadPlayer($user)) {
```

Comment:
Probably a comment, I would let GMT/NATs and other elevated users to check these private matches for moderation issues (checking cheaters in multiplayer rooms, etc.)

Reply:
admin already overrides everything.